sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -27,21 +28,28 @@ object StringUtils {
   // replace the % with .*, match 0 or more times with any character
   def escapeLikeRegex(v: String): String = {
```

Comment:
I think we could further simplify this by replacing `previousChar` with a boolean, `nextCharacterIsEscaped` (or `inEscape`).